Library/Homebrew/cask/cask_loader.rb

```diff
@@ -330,18 +336,17 @@ def from_h_hash_gsubs(hash, appdir)
         hash.to_h.transform_values do |value|
           from_h_gsubs(value, appdir)
         end
-      rescue TypeError
-        from_h_array_gsubs(hash, appdir)
       end
 
       def from_h_gsubs(value, appdir)
         return value if value.blank?
 
-        if value.respond_to? :to_h
+        case value
+        when Hash
           from_h_hash_gsubs(value, appdir)
-        elsif value.respond_to? :to_a
+        when Array
           from_h_array_gsubs(value, appdir)
-        elsif value.is_a? String
+        when String
```

Comment:
This was changed to better handle `uninstall signal` args. Essentially, `signal` takes arrays of two elements each which was getting turned into a hash here. Also, I think we can be more specific here since we know which data types we're dealing with (it's a JSON-like data structure).

https://docs.brew.sh/Cask-Cookbook#stanza-uninstall